File: 04-deberes/02-proyecto/AppZapatos/app/src/main/java/com/example/appzapatos/modulo_clientes/Cliente.kt
package com.example.appzapatos.modulo_clientes

class Cliente(
//    var compraDeCliente: ArrayList<Compra>?,
    var createdAt: Long?,
    var updatedAt: Long?,
    var id: Int?,
    var nombre: String,
    var cedula: String,
    var apellido: String
) {

}


//parcel.readSerializable() as? ArrayList<Compra>,
//parcel.writeSerializable(compraDeCliente)

File: 04-deberes/02-proyecto/Backend/zapatos/api/models/Cliente.js
/**
 * Cliente.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    nombre: {
      type: 'string',
      required: true,
      minLength: 3,
      maxLength: 15,
    },
    cedula: {
      type: 'string',
      required: true,
      unique: true,
      minLength: 10,
      maxLength: 25,
    },
    apellido: {
      type: 'string',
      required: true,
      minLength: 3,
      maxLength: 15,
    },
    compraDeCliente: {     // Nombre atributo de la relación
      collection: 'compra', // Nombre del modelo a relacionar
      via: 'codigoCli'        // Nombre del campo a hacer la relacion
    },
  },
};

